app/controllers/application_controller.rb

```diff
@@ -1,10 +1,33 @@
 require './config/environment'
+require "sinatra/reloader"
 
 class ApplicationController < Sinatra::Base
 
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    register Sinatra::Reloader
+    enable :reloader
+    enable :sessions
+    set :session_secret, "secret"
   end
 
+  get '/' do
+    erb :homepage
+  end
+
+  get '/users/:user_slug' do
+    @user = User.find_by_slug(params[:user_slug])
+    erb :"users/show"
+  end
+
+  helpers do
+    def logged_in?
+      !!session[:id]
+    end
+    def current_user
```

Comment:
Because in the lifespan of a request, you might call `current_user` multiple times, we'd prefer this method to only hit the database once and then 'cache' the user instance into an instance variable that could be returned on any subsequent call to `current_user`. You can accomplish this with Memoization.  

```ruby
def current_user
  @current_user ||= User.find(session[:id]) if session[:id]
end
```